.github/workflows/codeql.yaml

```diff
@@ -0,0 +1,35 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+  schedule:
+    - cron: "0 7 * * 1" # Mondays at 7:00 AM
+
+permissions: read-all
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@b398f525a5587552e573b247ac661067fafa920b
+        with:
+          languages: go
+
+      - name: Autobuild
+        uses: github/codeql-action/autobuild@b398f525a5587552e573b247ac661067fafa920b
+
+      - name: Perform CodeQL Analysis
```

Comment:
Can we visualize the output of this step? The action says it's uploading data.

Reply:
Any alerts from that run will be available in [Code Scanning](https://github.com/Azure/secrets-store-csi-driver-provider-azure/security/code-scanning)